src/bin/rgb.rs

```diff
@@ -382,11 +387,61 @@ fn main() -> Result<(), Error> {
             }
         },
         Command::Psbt { subcommand } => match subcommand {
-            PsbtCommand::Bundle { psbt_in, psbt_out } => {
+            PsbtCommand::Bundle {
+                psbt_in,
+                psbt_out,
+                method,
+            } => {
                 let psbt_bytes = fs::read(&psbt_in)?;
                 let mut psbt = Psbt::deserialize(&psbt_bytes)?;
 
-                let count = psbt.rgb_bundle_to_lnpbp4()?;
+                let mut count: usize = 0;
+                match method {
+                    CloseMethod::TapretFirst => {
+                        let mut first_tap_out: Option<usize> = None;
+                        for (i, o) in psbt.outputs.iter().enumerate() {
+                            if o.script.is_v1_p2tr() {
+                                first_tap_out = Some(i);
+                                break;
+                            }
+                        }
+                        if let Some(out_id) = first_tap_out {
+                            let output = psbt
+                                .outputs
+                                .get_mut(out_id)
+                                .expect("output should be in vector");
+                            if output.tapret_dfs_path().is_none() {
+                                output
+                                    .set_tapret_dfs_path(&DfsPath::from_str("1")?)
+                                    .expect("given output should be valid");
+                            }
+                        }
```

Comment:
This fragment can be re-written with more ideomatic rust without any exceptions/expects as:

```suggestion
                        if let Some(output) = psbt.outputs.iter_mut().find(Output::is_v1_p2tr) {
                          if output.tapret_dfs_path() {
                                output.set_tapret_dfs_path(&DfsPath::with([Dfsorder::First]))
                          }
                        }

```

Reply:
refactored as suggested, much better now, thanks for the tips

note: I didn't use `Output::is_v1_p2tr` as `is_v1_p2tr` is from bitcoin's `Script`; IMO the code is fine as it is now but feel free to correct me if I'm wrong